editoast/src/schema/electrical_profiles.rs

```diff
@@ -0,0 +1,200 @@
+    /// Couldn't found the infra with the given id
+    #[error("Electrical Profile Set '{0}', could not be found")]
+    NotFound(i64),
+    #[error("An internal diesel error occurred: '{}'", .0.to_string())]
+    DieselError(DieselError),
+    #[error("An internal error occurred: '{}'", .0.to_string())]
+    InternalError(String),
+}
+
+impl ApiError for ElectricalProfilesApiError {
+    fn get_status(&self) -> Status {
+        match self {
+            Self::NotFound(_) => Status::NotFound,
+            Self::DieselError(_) => Status::InternalServerError,
+            Self::InternalError(_) => Status::InternalServerError,
+        }
+    }
+
+    fn get_type(&self) -> &'static str {
+        match self {
+            Self::NotFound(_) => "editoast:electrical_profiles:NotFound",
+            Self::DieselError(_) => "editoast:electrical_profiles:DieselError",
+            Self::InternalError(_) => "editoast:electrical_profiles:InternalError",
+        }
+    }
+
+    fn extra(&self) -> Option<Map<String, Value>> {
+        match self {
+            ElectricalProfilesApiError::NotFound(electrical_profile_set_id) => json!({
+                "electrical_profile_set_id": electrical_profile_set_id,
+            })
+            .as_object()
+            .cloned(),
+            _ => None,
+        }
+    }
+}
+
+impl ElectricalProfileSetWrapper {
+    fn retrieve(
+        conn: &mut PgConnection,
+        ep_set_id: i64,
+    ) -> Result<ElectricalProfileSetWrapper, Box<dyn ApiError>> {
+        match dsl::osrd_infra_electricalprofileset
+            .find(ep_set_id)
+            .first(conn)
+        {
+            Ok(ep_set) => Ok(ep_set),
+            Err(DieselError::NotFound) => {
+                Err(Box::new(ElectricalProfilesApiError::NotFound(ep_set_id)))
+            }
+            Err(e) => Err(Box::new(ElectricalProfilesApiError::DieselError(e))),
+        }
+    }
+
+    pub fn retrieve_data(
+        conn: &mut PgConnection,
+        ep_set_id: i64,
+    ) -> Result<ElectricalProfileSet, Box<dyn ApiError>> {
+        let ep_set_wrapper = Self::retrieve(conn, ep_set_id);
+        match ep_set_wrapper {
+            Ok(ep_set_wrapper) => {
+                match serde_json::from_value::<ElectricalProfileSet>(ep_set_wrapper.data) {
+                    Ok(ep_set) => Ok(ep_set),
+                    Err(e) => Err(Box::new(ElectricalProfilesApiError::InternalError(
+                        e.to_string(),
+                    ))),
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn list(
+        conn: &mut PgConnection,
+    ) -> Result<Vec<ElectricalProfileSetMetaData>, Box<dyn ApiError>> {
+        match dsl::osrd_infra_electricalprofileset
+            .select((dsl::id, dsl::name))
+            .load(conn)
+        {
+            Ok(ep_sets) => Ok(ep_sets),
+            Err(e) => Err(Box::new(ElectricalProfilesApiError::DieselError(e))),
+        }
+    }
+}
```

Comment:
Queryable, errors and implementations should be moved to the view. 
The `schema` module is here to declare serializers.